scripts/validate-autoinstall-user-data.py

```diff
@@ -60,7 +60,13 @@ def main() -> None:
         assert user_data.readline() == "#cloud-config\n"
         def get_autoinstall_data(data): return data["autoinstall"]
     else:
-        def get_autoinstall_data(data): return data
+        def get_autoinstall_data(data):
+            try:
+                cfg = data.pop("autoinstall")
```

Comment:
I guess it's better not to mutate the original object if we return the data we're interested in? I'd suggest

```python
cfg = data["autoinstall"]
```

Reply:
Ah good point.  I think in this script the way we consume the config it wouldn't be a deal breaker but I agree in principle. I get the KeyError I wanted using `[]` anyways. Updated, thanks!